In [38]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np
import scipy.stats as stats

In [ ]:
# 출력물 잘리지 않게 display option 설정
pd.set_option("display.max_rows", None)
pd.set_option("display.max_columns", None)
pd.set_option("display.width", None)
pd.set_option("display.max_colwidth", None)

In [20]:
df = pd.read_csv("../data/train.csv")

In [21]:
df

ID 시술 시기 코드 시술 당시 나이  임신 시도 또는 마지막 임신 경과 연수 시술 유형 특정 시술 유형  \
0       TRAIN_000000   TRZKPL  만18-34세                    NaN   IVF     ICSI   
1       TRAIN_000001   TRYBLT  만45-50세                    NaN   IVF     ICSI   
2       TRAIN_000002   TRVNRY  만18-34세                    NaN   IVF      IVF   
3       TRAIN_000003   TRJXFG  만35-37세                    NaN   IVF     ICSI   
4       TRAIN_000004   TRVNRY  만18-34세                    NaN   IVF     ICSI   
...              ...      ...      ...                    ...   ...      ...   
256346  TRAIN_256346   TRYBLT  만18-34세                    NaN   IVF     ICSI   
256347  TRAIN_256347   TRYBLT  만38-39세                    NaN   IVF     ICSI   
256348  TRAIN_256348   TRVNRY  만35-37세                    NaN   IVF     ICSI   
256349  TRAIN_256349   TRZKPL  만38-39세                    NaN   IVF     ICSI   
256350  TRAIN_256350   TRXQMD  만35-37세                    NaN   IVF      IVF   

        배란 자극 여부    배란 유도 유형  단일 배아 이식 여부  착상 전 유전 검사 사용 여부  ...  기증 배아 사용 여부  \
0              1  기록되지 않은 시행          0.0               NaN  ...          0.0   
1              0      알 수 없음          0.0               NaN  ...          0.0   
2              1  기록되지 않은 시행          0.0               NaN  ...          0.0   
3              1  기록되지 않은 시행          0.0               NaN  ...          0.0   
4              1  기록되지 않은 시행          0.0               NaN  ...          0.0   
...          ...         ...          ...               ...  ...          ...   
256346         1  기록되지 않은 시행          0.0               NaN  ...          0.0   
256347         1  기록되지 않은 시행          0.0               NaN  ...          0.0   
256348         1  기록되지 않은 시행          0.0               NaN  ...          0.0   
256349         1  기록되지 않은 시행          0.0               NaN  ...          0.0   
256350         0      알 수 없음          0.0               NaN  ...          0.0   

        대리모 여부  PGD 시술 여부  PGS 시술 여부  난자 채취 경과일  난자 해동 경과일  난자 혼합 경과일  \
0          0.0        NaN        NaN        0.0        NaN        0.0   
1          0.0        NaN        NaN        0.0        NaN        0.0   
2          0.0        NaN        NaN        0.0        NaN        0.0   
3          0.0        NaN        NaN        0.0        NaN        0.0   
4          0.0        NaN        NaN        0.0        NaN        0.0   
...        ...        ...        ...        ...        ...        ...   
256346     0.0        NaN        NaN        0.0        NaN        0.0   
256347     0.0        NaN        NaN        0.0        NaN        0.0   
256348     0.0        NaN        NaN        0.0        NaN        0.0   
256349     0.0        NaN        NaN        0.0        NaN        0.0   
256350     0.0        NaN        NaN        NaN        NaN        NaN   

        배아 이식 경과일  배아 해동 경과일  임신 성공 여부  
0             3.0        NaN         0  
1             NaN        NaN         0  
2             2.0        NaN         0  
3             NaN        NaN         0  
4             3.0        NaN         0  
...           ...        ...       ...  
256346        5.0        NaN         0  
256347        3.0        NaN         1  
256348        3.0        NaN         0  
256349        1.0        NaN         1  
256350        0.0        0.0         1  

[256351 rows x 69 columns]

### 1. 데이터 분포 확인

In [22]:
# 데이터 정보
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 256351 entries, 0 to 256350
Data columns (total 69 columns):
 #   Column                 Non-Null Count   Dtype  
---  ------                 --------------   -----  
 0   ID                     256351 non-null  str    
 1   시술 시기 코드               256351 non-null  str    
 2   시술 당시 나이               256351 non-null  str    
 3   임신 시도 또는 마지막 임신 경과 연수  9370 non-null    float64
 4   시술 유형                  256351 non-null  str    
 5   특정 시술 유형               256349 non-null  str    
 6   배란 자극 여부               256351 non-null  int64  
 7   배란 유도 유형               256351 non-null  str    
 8   단일 배아 이식 여부            250060 non-null  float64
 9   착상 전 유전 검사 사용 여부       2718 non-null    float64
 10  착상 전 유전 진단 사용 여부       250060 non-null  float64
 11  남성 주 불임 원인             256351 non-null  int64  
 12  남성 부 불임 원인             256351 non-null  int64  
 13  여성 주 불임 원인             256351 non-null  int64  
 14  여성 부 불임 원인             256351 non-null  int64  

In [33]:
# 결측치 확인
df.isnull().sum()

ID                            0
시술 시기 코드                      0
시술 당시 나이                      0
임신 시도 또는 마지막 임신 경과 연수    246981
시술 유형                         0
특정 시술 유형                      2
배란 자극 여부                      0
배란 유도 유형                      0
단일 배아 이식 여부                6291
착상 전 유전 검사 사용 여부         253633
착상 전 유전 진단 사용 여부           6291
남성 주 불임 원인                    0
남성 부 불임 원인                    0
여성 주 불임 원인                    0
여성 부 불임 원인                    0
부부 주 불임 원인                    0
부부 부 불임 원인                    0
불명확 불임 원인                     0
불임 원인 - 난관 질환                 0
불임 원인 - 남성 요인                 0
불임 원인 - 배란 장애                 0
불임 원인 - 여성 요인                 0
불임 원인 - 자궁경부 문제               0
불임 원인 - 자궁내막증                 0
불임 원인 - 정자 농도                 0
불임 원인 - 정자 면역학적 요인            0
불임 원인 - 정자 운동성                0
불임 원인 - 정자 형태                 0
배아 생성 주요 이유                6291
총 시술 횟수                       0
클리닉 내 총 시술 횟수                 0
IVF 시술 횟

In [24]:
# 결측치 많은 컬럼 확인
df.isnull().mean().sort_values(ascending=False).head(10)

난자 해동 경과일                0.994398
PGS 시술 여부                0.992475
PGD 시술 여부                0.991500
착상 전 유전 검사 사용 여부         0.989397
임신 시도 또는 마지막 임신 경과 연수    0.963449
배아 해동 경과일                0.842525
난자 채취 경과일                0.224255
난자 혼합 경과일                0.209615
배아 이식 경과일                0.169947
총 생성 배아 수                0.024541
dtype: float64

In [34]:
# 전체 변수 결측 보기 
missing_summary = (
    df.isnull()
    .mean()
    .sort_values(ascending=False)
    .to_frame(name="missing_ratio")
)

missing_summary["missing_percent"] = (missing_summary["missing_ratio"])

missing_summary

missing_ratio  missing_percent
난자 해동 경과일                   0.994398         0.994398
PGS 시술 여부                   0.992475         0.992475
PGD 시술 여부                   0.991500         0.991500
착상 전 유전 검사 사용 여부            0.989397         0.989397
임신 시도 또는 마지막 임신 경과 연수       0.963449         0.963449
배아 해동 경과일                   0.842525         0.842525
난자 채취 경과일                   0.224255         0.224255
난자 혼합 경과일                   0.209615         0.209615
배아 이식 경과일                   0.169947         0.169947
총 생성 배아 수                   0.024541         0.024541
저장된 신선 난자 수                 0.024541         0.024541
기증자 정자와 혼합된 난자 수            0.024541         0.024541
파트너 정자와 혼합된 난자 수            0.024541         0.024541
혼합된 난자 수                    0.024541         0.024541
해동 난자 수                     0.024541         0.024541
수집된 신선 난자 수                 0.024541         0.024541
이식된 배아 수                    0.024541         0.024541
배아 생성 주요 이유                 0.024541         0.024541
해동된 배아 수                    0.024541         0.024541
미세주입 배아 이식 수                0.024541         0.024541
기증 배아 사용 여부                 0.024541         0.024541
미세주입에서 생성된 배아 수             0.024541         0.024541
미세주입된 난자 수                  0.024541         0.024541
동결 배아 사용 여부                 0.024541         0.024541
신선 배아 사용 여부                 0.024541         0.024541
저장된 배아 수                    0.024541         0.024541
대리모 여부                      0.024541         0.024541
착상 전 유전 진단 사용 여부            0.024541         0.024541
단일 배아 이식 여부                 0.024541         0.024541
미세주입 후 저장된 배아 수             0.024541         0.024541
특정 시술 유형                    0.000008         0.000008
ID                          0.000000         0.000000
난자 출처                       0.000000         0.000000
정자 출처                       0.000000         0.000000
난자 기증자 나이                   0.000000         0.000000
DI 출산 횟수                    0.000000         0.000000
정자 기증자 나이                   0.000000         0.000000
IVF 임신 횟수                   0.000000         0.000000
IVF 출산 횟수                   0.000000         0.000000
불임 원인 - 배란 장애               0.000000         0.000000
시술 당시 나이                    0.000000         0.000000
시술 유형                       0.000000         0.000000
배란 자극 여부                    0.000000         0.000000
배란 유도 유형                    0.000000         0.000000
남성 주 불임 원인                  0.000000         0.000000
남성 부 불임 원인                  0.000000         0.000000
여성 주 불임 원인                  0.000000         0.000000
여성 부 불임 원인                  0.000000         0.000000
부부 주 불임 원인                  0.000000         0.000000
부부 부 불임 원인                  0.000000         0.000000
불명확 불임 원인                   0.000000         0.000000
불임 원인 - 난관 질환               0.000000         0.000000
불임 원인 - 남성 요인               0.000000         0.000000
불임 원인 - 여성 요인               0.000000         0.000000
총 출산 횟수                     0.000000         0.000000
불임 원인 - 자궁경부 문제             0.000000         0.000000
불임 원인 - 자궁내막증               0.000000         0.000000
불임 원인 - 정자 농도               0.000000         0.000000
불임 원인 - 정자 면역학적 요인          0.000000         0.000000
불임 원인 - 정자 운동성              0.000000         0.000000
불임 원인 - 정자 형태               0.000000         0.000000
총 시술 횟수                     0.000000         0.000000
클리닉 내 총 시술 횟수               0.000000         0.000000
IVF 시술 횟수                   0.000000         0.000000
DI 시술 횟수                    0.000000         0.000000
총 임신 횟수                     0.000000         0.000000
시술 시기 코드                    0.000000         0.000000
DI 임신 횟수                    0.000000         0.000000
임신 성공 여부                    0.000000         0.000000

In [25]:
# columns 확인
list(df.columns)

['ID',
 '시술 시기 코드',
 '시술 당시 나이',
 '임신 시도 또는 마지막 임신 경과 연수',
 '시술 유형',
 '특정 시술 유형',
 '배란 자극 여부',
 '배란 유도 유형',
 '단일 배아 이식 여부',
 '착상 전 유전 검사 사용 여부',
 '착상 전 유전 진단 사용 여부',
 '남성 주 불임 원인',
 '남성 부 불임 원인',
 '여성 주 불임 원인',
 '여성 부 불임 원인',
 '부부 주 불임 원인',
 '부부 부 불임 원인',
 '불명확 불임 원인',
 '불임 원인 - 난관 질환',
 '불임 원인 - 남성 요인',
 '불임 원인 - 배란 장애',
 '불임 원인 - 여성 요인',
 '불임 원인 - 자궁경부 문제',
 '불임 원인 - 자궁내막증',
 '불임 원인 - 정자 농도',
 '불임 원인 - 정자 면역학적 요인',
 '불임 원인 - 정자 운동성',
 '불임 원인 - 정자 형태',
 '배아 생성 주요 이유',
 '총 시술 횟수',
 '클리닉 내 총 시술 횟수',
 'IVF 시술 횟수',
 'DI 시술 횟수',
 '총 임신 횟수',
 'IVF 임신 횟수',
 'DI 임신 횟수',
 '총 출산 횟수',
 'IVF 출산 횟수',
 'DI 출산 횟수',
 '총 생성 배아 수',
 '미세주입된 난자 수',
 '미세주입에서 생성된 배아 수',
 '이식된 배아 수',
 '미세주입 배아 이식 수',
 '저장된 배아 수',
 '미세주입 후 저장된 배아 수',
 '해동된 배아 수',
 '해동 난자 수',
 '수집된 신선 난자 수',
 '저장된 신선 난자 수',
 '혼합된 난자 수',
 '파트너 정자와 혼합된 난자 수',
 '기증자 정자와 혼합된 난자 수',
 '난자 출처',
 '정자 출처',
 '난자 기증자 나이',
 '정자 기증자 나이',
 '동결 배아 사용 여부',
 '신선 배아 사용 여부',
 '기증 배아 사용 여부',
 '대리모 여부',
 'PGD 시술 여부',
 'PGS 시술 여부',
 '난자 채취 경과일',
 '난자 해동 경과일',
 '난자

In [26]:
# 타겟 분포 확인
df["임신 성공 여부"].value_counts(normalize=True)

임신 성공 여부
0    0.741651
1    0.258349
Name: proportion, dtype: float64

난임 환자 256,351명 중 25%정도만 임신 성공

결측이 많은 이유는 해당 시술을 한 사람만 응답을 하기 때문
ex) 난자 해동 경과일 결측 0.994398 = 난자 해동을 한 사람이 0.056% 정도밖에 안됨

-----------------------------------

### 2. 특정 시술 유형

In [35]:
df["특정 시술 유형"].value_counts()

특정 시술 유형
ICSI                                   122368
IVF                                     91755
Unknown                                 26939
IUI                                      6100
ICSI:ICSI                                2314
ICSI / BLASTOCYST                        1609
IVF / BLASTOCYST                         1248
IVF:IVF                                  1146
ICSI:IVF                                  873
ICSI / AH                                 769
IVF:ICSI                                  392
IVF / AH                                  319
ICSI:Unknown                              207
IVF:Unknown                               100
ICI                                        96
Generic DI                                 71
IVI                                        23
ICSI / BLASTOCYST:IVF / BLASTOCYST          6
ICSI / BLASTOCYST :IVF / BLASTOCYST         4
FER                                         3
IVF / AH:ICSI / AH                          2
GIFT                     

#### 대분류로 구분해두고 target 비교

In [37]:
def major_procedure(x):
    if pd.isna(x):
        return "Unknown"
    if "IUI" in x:
        return "IUI"
    if "DI" in x:
        return "DI"
    if "ICSI" in x:
        return "ICSI"
    if "IVF" in x:
        return "IVF"
    if "FER" in x:
        return "FER"
    if "GIFT" in x:
        return "GIFT"
    return "Other"

df["시술_대분류"] = df["특정 시술 유형"].apply(major_procedure)

rate = (
    df.groupby("시술_대분류")["임신 성공 여부"]
      .agg(["count", "mean"])
      .rename(columns={"mean": "success_rate"})
      .sort_values("success_rate", ascending=False)
)

rate

count  success_rate
시술_대분류                       
FER           3      0.333333
ICSI     128547      0.268314
IVF       94568      0.259443
Other     27058      0.236714
DI           71      0.169014
IUI        6100      0.128525
GIFT          2      0.000000
Unknown       2      0.000000

In [43]:
# FER, GIFT, DI, Unknown sample수가 너무 적음 -> Other로 통합
rare = ["FER", "GIFT", "Unknown", "DI"]
df["시술_대분류"] = df["시술_대분류"].replace(rare, "Other")

In [47]:
# 합치고 다시 확인
rate = (
    df.groupby("시술_대분류")["임신 성공 여부"]
      .agg(["count", "mean"])
      .rename(columns={"mean": "success_rate"})
      .sort_values("success_rate", ascending=False)
)

rate

count  success_rate
시술_대분류                      
ICSI    128547      0.268314
IVF      94568      0.259443
Other    27136      0.236512
IUI       6100      0.128525

In [46]:
# 통계적으로 차이가 있는지 (p<0.05)
ct = pd.crosstab(df["시술_대분류"], df["임신 성공 여부"])
chi2, p, _, _ = stats.chi2_contingency(ct)
print(f"chi2 p-value: {p:.3e}")

chi2 p-value: 3.456e-145


In [ ]:
# 세부적으로 (Blastocyst | AH)
df["BLASTOCYST_포함"] = df["특정 시술 유형"].str.contains("BLASTOCYST", na=False).astype(int)
df["AH_포함"]         = df["특정 시술 유형"].str.contains("AH", na=False).astype(int)

In [ ]:
# Blastocyst 여부에 따른 임신 성공 여부 차이 
df.groupby("BLASTOCYST_포함")["임신 성공 여부"] \
  .agg(["count", "mean"]) \
  .rename(columns={"mean": "success_rate"})

count  success_rate
BLASTOCYST_포함                      
0              253483      0.257189
1                2868      0.360879

In [51]:
# AH 여부에 따른 임신 성공 여부 차이
df.groupby("AH_포함")["임신 성공 여부"] \
  .agg(["count", "mean"]) \
  .rename(columns={"mean": "success_rate"})

count  success_rate
AH_포함                      
0      255259      0.258565
1        1092      0.207875

##### 배아 시술 여부와 밑에서 만든 배아 이식여부 간 겹치는 데이터가 얼마나 있는지

In [ ]:
df["배아_시술_여부"] = df["시술_대분류"].isin(["IVF", "ICSI", "FER"]).astype(int)

pd.crosstab(
    df["배아_시술_여부"],
    df["배아_이식_여부"],
    margins=True
)

배아_이식_여부     0       1     All
배아_시술_여부                      
0         6291   26945   33236
1            0  223115  223115
All       6291  250060  256351

+ 배아 이식 시술을 받은 사람이 22만명,,, 전체 표본의 대부분이 배아 기반 시술을 받음

-----------------------------

#### 3. 배아 관련 변수 결측치 6291 = 배아 이식 미도달

In [53]:
embryo_stage_cols = [
    "단일 배아 이식 여부",
    "착상 전 유전 진단 사용 여부",
    "배아 생성 주요 이유",
    "총 생성 배아 수",
    "미세주입된 난자 수",
    "미세주입에서 생성된 배아 수",
    "이식된 배아 수",
    "미세주입 배아 이식 수",
    "저장된 배아 수",
    "미세주입 후 저장된 배아 수",
    "해동된 배아 수",
    "해동 난자 수",
    "수집된 신선 난자 수",
    "저장된 신선 난자 수",
    "혼합된 난자 수",
    "파트너 정자와 혼합된 난자 수",
    "기증자 정자와 혼합된 난자 수",
    "동결 배아 사용 여부",
    "신선 배아 사용 여부",
    "기증 배아 사용 여부",
    "대리모 여부",
]

df["배아_이식_미도달"] = df[embryo_stage_cols].isna().all(axis=1).astype(int)

df["배아_이식_미도달"].sum()

np.int64(6291)

In [ ]:
# 배아 이식 여부 (0/1)로 바꿔줌
df["배아_이식_여부"] = 1 - df["배아_이식_미도달"]

In [58]:
pd.crosstab(df["배아_이식_여부"], df["임신 성공 여부"])

임신 성공 여부       0      1
배아_이식_여부               
0           5480    811
1         184643  65417

----------------

#### 5. 불임 원인

In [67]:
infertility_cols = [
    "남성 주 불임 원인",
    "남성 부 불임 원인",
    "여성 주 불임 원인",
    "여성 부 불임 원인",
    "부부 주 불임 원인",
    "부부 부 불임 원인",
    "불명확 불임 원인",
    "불임 원인 - 난관 질환",
    "불임 원인 - 남성 요인",
    "불임 원인 - 배란 장애",
    "불임 원인 - 여성 요인",
    "불임 원인 - 자궁경부 문제",
    "불임 원인 - 자궁내막증",
    "불임 원인 - 정자 농도",
    "불임 원인 - 정자 면역학적 요인",
    "불임 원인 - 정자 운동성",
    "불임 원인 - 정자 형태",
]

df["불임_원인_개수"] = df[infertility_cols].sum(axis=1)

df.groupby("불임_원인_개수")["임신 성공 여부"] \
  .agg(["count", "mean"]) \
  .rename(columns={"mean": "success_rate"})

count  success_rate
불임_원인_개수                      
0          44310      0.204694
1         169046      0.271719
2          30033      0.278028
3           2750      0.247636
4           8654      0.214698
5           1371      0.218818
6            137      0.189781
7             43      0.232558
8              7      0.000000

In [68]:
df.groupby("임신 성공 여부")["불임_원인_개수"].mean()

임신 성공 여부
0    1.084445
1    1.114846
Name: 불임_원인_개수, dtype: float64

원인이 3개 이상 있을 경우부터 성공률이 줄어들긴 하지만 임신 성공 확률에 큰 영향을 미치지는 않음

-----------

#### 나이/기간 관련

In [75]:
df.groupby("정자 기증자 나이")["임신 성공 여부"] \
  .agg(["count", "mean"]) \
  .rename(columns={"mean": "success_rate"})

count  success_rate
정자 기증자 나이                      
만20세 이하      1067      0.209934
만21-25세      5667      0.241398
만26-30세      5058      0.248122
만31-35세      4911      0.248626
만36-40세      5282      0.245740
만41-45세      3848      0.242983
알 수 없음     230518      0.259967

In [76]:
df.groupby("난자 기증자 나이")["임신 성공 여부"] \
  .agg(["count", "mean"]) \
  .rename(columns={"mean": "success_rate"})

count  success_rate
난자 기증자 나이                      
만20세 이하       294      0.261905
만21-25세      2334      0.329906
만26-30세      4976      0.348272
만31-35세      6366      0.305215
알 수 없음     242381      0.254579

In [77]:
df.groupby("시술 당시 나이")["임신 성공 여부"] \
  .agg(["count", "mean"]) \
  .rename(columns={"mean": "success_rate"})

count  success_rate
시술 당시 나이                      
만18-34세   102476      0.322622
만35-37세    57780      0.278401
만38-39세    39247      0.217138
만40-42세    37348      0.159393
만43-44세    12253      0.118012
만45-50세     6918      0.167679
알 수 없음       329      0.000000

In [ ]:
df.groupby("총 시술 횟수")["임신 성공 여부"] \
  .agg(["count", "mean"]) \
  .rename(columns={"mean": "success_rate"})

# 시술 횟수에 따른 차이도 존재함

count  success_rate
총 시술 횟수                     
0회       97599      0.290987
1회       56819      0.249952
2회       39338      0.244166
3회       24531      0.236354
4회       15141      0.227792
5회        9106      0.215792
6회 이상    13817      0.203300

In [82]:
def collapse_trials(x):
    if x == '0회':
        return '0회'
    elif x in ['1회', '2회']:
        return '1–2회'
    else:
        return '3회 이상'

df["총시술_bin3"] = df["총 시술 횟수"].apply(collapse_trials)

rate3 = (
    df.groupby("총시술_bin3")["임신 성공 여부"]
      .agg(["count", "mean"])
      .rename(columns={"mean": "success_rate"})
)

rate3


count  success_rate
총시술_bin3                     
0회        97599      0.290987
1–2회      96157      0.247585
3회 이상     62595      0.223996

In [83]:
df.groupby("이식된 배아 수")["임신 성공 여부"] \
  .agg(["count", "mean"]) \
  .rename(columns={"mean": "success_rate"})

count  success_rate
이식된 배아 수                      
0.0        36544      0.000821
1.0        93791      0.313548
2.0       110845      0.311092
3.0         8880      0.168468

In [86]:
df.groupby("배아 이식 경과일")["임신 성공 여부"] \
  .agg(["count", "mean"]) \
  .rename(columns={"mean": "success_rate"})

count  success_rate
배아 이식 경과일                     
0.0        24904      0.251044
1.0         6053      0.187015
2.0        35078      0.212469
3.0        57924      0.258770
4.0         4504      0.344361
5.0        81459      0.404449
6.0         2773      0.300036
7.0           90      0.411111

In [89]:
df.groupby("난자 채취 경과일")["임신 성공 여부"] \
  .agg(["count", "mean"]) \
  .rename(columns={"mean": "success_rate"})

count  success_rate
난자 채취 경과일                      
0.0        198863      0.263815

In [90]:
df.groupby("난자 해동 경과일")["임신 성공 여부"] \
  .agg(["count", "mean"]) \
  .rename(columns={"mean": "success_rate"})

count  success_rate
난자 해동 경과일                     
0.0         1434      0.222455
1.0            2      0.000000

In [91]:
df.groupby("난자 혼합 경과일")["임신 성공 여부"] \
  .agg(["count", "mean"]) \
  .rename(columns={"mean": "success_rate"})

count  success_rate
난자 혼합 경과일                      
0.0        201920      0.277075
1.0           488      0.153689
2.0           102      0.245098
3.0            68      0.279412
4.0             5      0.400000
5.0            24      0.208333
6.0             8      0.000000
7.0             1      0.000000